controllers/secretdefinition_controller.go

```diff
@@ -216,10 +200,42 @@ func (r *SecretDefinitionReconciler) AddFinalizerIfNotPresent(sDef *smv1alpha1.S
 	return nil
 }
 
-// +kubebuilder:rbac:groups=secrets-manager.tuenti.io,resources=secretdefinitions,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups=secrets-manager.tuenti.io,resources=secretdefinitions/status,verbs=get;update;patch
-// +kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch;create;update;patch;delete
-func (r *SecretDefinitionReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+// Helper functions to manage corev1.Secret and smv1alpha1.SecretDefinition
+func getObjectMetaFromSecretDefinition(sDef *smv1alpha1.SecretDefinition) metav1.ObjectMeta {
+	labels := map[string]string{
+		managedByLabel: "secrets-manager",
+	}
+	annotations := map[string]string{
+		lastUpdateLabel: time.Now().Format(timestampFormat),
+	}
+
+	mergeMap(labels, sDef.Labels, noSkip)
+	mergeMap(annotations, sDef.Annotations, skipAnnotation)
+
+	return metav1.ObjectMeta{
+		Namespace:   sDef.Namespace,
+		Name:        sDef.Spec.Name,
+		Labels:      labels,
+		Annotations: annotations,
+	}
+}
+
+//+kubebuilder:rbac:groups=secrets-manager.tuenti.io,resources=secretdefinitions,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=secrets-manager.tuenti.io,resources=secretdefinitions/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=secrets-manager.tuenti.io,resources=secretdefinitions/finalizers,verbs=update
+//+kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the SecretDefinition object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
+func (r *SecretDefinitionReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
```

Comment:
With the new version of `kubebuilder` a context is passed to the `Reconcile` method, nevertheless you keep using the context associated with the `Reconciler`. I think we should use the passed context and remove the `Ctx` attribute from the `SecretDefinitionReconciler`

Reply:
Context is now used that is passed from the reconcile function in all other functions.